non-existing package should give a better exception

When using the wrong package name, the error produced looked like this:

```
[ERROR] Failed to execute goal com.github.havardh:javaflow-maven-plugin:1.0.0:build (default-cli) on project cap: Could not generate types: Parameter 'directory' is not a directory -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
```

This turned out to be because the package defined in `packageName` was incorrect, thus generating an invalid `baseSourceDirectory` in https://github.com/havardh/javaflow-maven-plugin/blob/ae5969a65c252e73d5ca06f0be55332e1a3cf320/src/main/java/com/github/havardh/javaflow/plugins/JavaflowMojo.java#L56

Contents of the paramter 'directory' could be useful information

--- a/src/main/java/com/github/havardh/javaflow/plugins/Api.java
+++ b/src/main/java/com/github/havardh/javaflow/plugins/Api.java
@@ -1,7 +1,13 @@
 package com.github.havardh.javaflow.plugins;
 
+import static java.util.Collections.emptyMap;
+import static java.util.Collections.singletonList;
+import static java.util.Optional.ofNullable;
+
 import java.util.List;
 import java.util.Map;
+
+import com.github.havardh.javaflow.plugins.exceptions.MissingPackageParameter;
 
 public class Api {
 
@@ -13,18 +19,18 @@ public class Api {
   public Api() {}
 
   public String getPackageName() {
-    return packageName;
+    return ofNullable(packageName).orElseThrow(MissingPackageParameter::new);
   }
 
   public List<String> getSuffixes() {
-    return suffixes;
+    return ofNullable(suffixes).orElse(singletonList(".java"));
   }
 
   public String getOutput() {
-    return output;
+    return ofNullable(output).orElse("flow.js");
   }
 
   public Map<String, String> getTypes() {
-    return types;
+    return ofNullable(types).orElse(emptyMap());
   }
 }

--- a/src/main/java/com/github/havardh/javaflow/plugins/JavaflowMojo.java
+++ b/src/main/java/com/github/havardh/javaflow/plugins/JavaflowMojo.java
@@ -31,6 +31,7 @@ import com.github.havardh.javaflow.phases.verifier.MemberFieldsPresentVerifier;
 import com.github.havardh.javaflow.phases.writer.flow.FlowWriter;
 import com.github.havardh.javaflow.phases.writer.flow.converter.Converter;
 import com.github.havardh.javaflow.phases.writer.flow.converter.JavaFlowConverter;
+import com.github.havardh.javaflow.plugins.exceptions.PackageDirectoryNotFound;
 
 /**
  * Generates flow types for a set of java models
@@ -64,7 +65,7 @@ public class JavaflowMojo extends AbstractMojo {
     try {
       apis.forEach(this::run);
     } catch (Exception e) {
-      throw new MojoExecutionException("Could not generate types", e);
+      throw new MojoExecutionException("Could not generate types\n\n", e);
     }
   }
 
@@ -75,6 +76,12 @@ public class JavaflowMojo extends AbstractMojo {
    */
   private void run(Api api) {
     String baseSourceDirectory = sourceDirectory + "/" + api.getPackageName().replace('.', '/');
+
+    File baseDirectoryFile = new File(baseSourceDirectory);
+
+    if (!baseDirectoryFile.isDirectory()) {
+      throw new PackageDirectoryNotFound(api.getPackageName(), baseSourceDirectory);
+    }
 
     Collection<File> files = FileUtils.listFiles(
         new File(baseSourceDirectory),

--- a/src/main/java/com/github/havardh/javaflow/plugins/JavaflowMojo.java
+++ b/src/main/java/com/github/havardh/javaflow/plugins/JavaflowMojo.java
@@ -75,19 +75,7 @@ public class JavaflowMojo extends AbstractMojo {
    * @param api - the api to generate flowtypes for
    */
   private void run(Api api) {
-    String baseSourceDirectory = sourceDirectory + "/" + api.getPackageName().replace('.', '/');
-
-    File baseDirectoryFile = new File(baseSourceDirectory);
-
-    if (!baseDirectoryFile.isDirectory()) {
-      throw new PackageDirectoryNotFound(api.getPackageName(), baseSourceDirectory);
-    }
-
-    Collection<File> files = FileUtils.listFiles(
-        new File(baseSourceDirectory),
-        new SuffixFileFilter(api.getSuffixes().toArray(new String[]{})),
-        TrueFileFilter.INSTANCE
-    );
+    Collection<File> files = getModelFiles(api);
 
     TypeMap typeMap = api.getTypes() == null ? TypeMap.emptyTypeMap() : new TypeMap(api.getTypes());
 
@@ -126,5 +114,21 @@ public class JavaflowMojo extends AbstractMojo {
     } catch (IOException e) {
       getLog().error(e);
     }
+  }
+
+  private Collection<File> getModelFiles(Api api) {
+    String baseSourceDirectory = sourceDirectory + "/" + api.getPackageName().replace('.', '/');
+
+    File baseDirectoryFile = new File(baseSourceDirectory);
+
+    if (!baseDirectoryFile.isDirectory()) {
+      throw new PackageDirectoryNotFound(api.getPackageName(), baseSourceDirectory);
+    }
+
+    return FileUtils.listFiles(
+        new File(baseSourceDirectory),
+        new SuffixFileFilter(api.getSuffixes().toArray(new String[]{})),
+        TrueFileFilter.INSTANCE
+    );
   }
 }

--- a/src/main/java/com/github/havardh/javaflow/plugins/JavaflowMojo.java
+++ b/src/main/java/com/github/havardh/javaflow/plugins/JavaflowMojo.java
@@ -1,5 +1,6 @@
 package com.github.havardh.javaflow.plugins;
 
+import static java.lang.String.format;
 import static java.util.Arrays.asList;
 import static java.util.Collections.singletonList;
 
@@ -110,7 +111,9 @@ public class JavaflowMojo extends AbstractMojo {
     );
 
     try {
-      Files.write(Paths.get(targetDirectory + "/" + api.getOutput()), asList(flow.split("\n")));
+      String outputFile = targetDirectory + "/" + api.getOutput();
+      Files.write(Paths.get(outputFile), asList(flow.split("\n")));
+      getLog().info(format("Wrote %d types to %s.", files.size(), outputFile));
     } catch (IOException e) {
       getLog().error(e);
     }

--- a/src/main/java/com/github/havardh/javaflow/plugins/exceptions/MissingPackageParameter.java
+++ b/src/main/java/com/github/havardh/javaflow/plugins/exceptions/MissingPackageParameter.java
@@ -1,0 +1,11 @@
+package com.github.havardh.javaflow.plugins.exceptions;
+
+public class MissingPackageParameter extends RuntimeException {
+
+  /**
+   * Create exception for missing package parameter
+   */
+  public MissingPackageParameter() {
+    super("Missing parameter apis.api.packageName");
+  }
+}

--- a/src/main/java/com/github/havardh/javaflow/plugins/exceptions/PackageDirectoryNotFound.java
+++ b/src/main/java/com/github/havardh/javaflow/plugins/exceptions/PackageDirectoryNotFound.java
@@ -1,0 +1,16 @@
+package com.github.havardh.javaflow.plugins.exceptions;
+
+import static java.lang.String.format;
+
+public class PackageDirectoryNotFound extends RuntimeException {
+
+  public PackageDirectoryNotFound(String packageName, String folder) {
+    super(format(
+        "Base directory for package '%s' was not found. \n" +
+        "Expected to find it here: %s",
+        packageName,
+        folder
+    ));
+  }
+
+}

